src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/ReadFilterArgumentCollection.java

```diff
@@ -0,0 +1,251 @@
+                    args.readGroupBlackListFilter.blackList, hdr);
+            // update the actual argument collection filter just in case...
+            args.readGroupBlackListFilter = rgblf;
+            return rgblf;
+        }),
+        READ_NAME((hdr, args) -> {
+            validateArgs(CommandLineReadFilter.valueOf("READ_NAME"), args.readNameFilter::validate);
+            return args.readNameFilter;
+        }),
+        READ_LENGTH((hdr, args) -> {
+            validateArgs(CommandLineReadFilter.valueOf("READ_LENGTH"), args.readLengthFilter::validate);
+            return args.readLengthFilter;
+        }),
+        READ_STRAND((hdr, args) -> { return args.readStrandFilter; }),
+        SAMPLE((hdr, args) -> {
+            validateArgs(CommandLineReadFilter.valueOf("SAMPLE"), args.sampleFilter::validate);
+            SampleReadFilter sf = new SampleReadFilter(hdr);
+            sf.samplesToKeep = args.sampleFilter.samplesToKeep;
+            // update the actual argument collection filter just in case...
+            args.sampleFilter = sf;
+            return sf;
+        }),
+        WELLFORMED((hdr, args) -> new WellformedReadFilter(hdr));
+
+        /**
+         * Each filter in the enum has an accompanying filter supplier function that accepts
+         * a ReadFilterArgumentCollection object, extracts the filter's required arguments
+         * from the argument collection, validates the arguments, and returns the initialized
+         * read filter.
+         */
+        private BiFunction<SAMFileHeader, ReadFilterArgumentCollection, ReadFilter> filterSupplier = null;
+        CommandLineReadFilter(final BiFunction<SAMFileHeader, ReadFilterArgumentCollection,ReadFilter> filterSupplier) {
+            this.filterSupplier = filterSupplier;
+        }
+
+        /**
+         * Execute the filter supplier and return the initialized filter
+         *
+         */
+        public ReadFilter getFilterInstance(
+                final SAMFileHeader samHeader,
+                final ReadFilterArgumentCollection readFilterCollection)
+        {
+            return filterSupplier.apply(samHeader, readFilterCollection);
+        };
+
+        public static boolean validateArgs(
+                final CommandLineReadFilter filter,
+                final Supplier<String> validator)
+        {
+            String errorMessage = validator.get();
+            if (null != errorMessage) {
+                throw new UserException.CommandLineException("The " + filter.name() + " read filter " + errorMessage);
+            }
+            return true;
+        }
+    }
+
+    @Argument(fullName = StandardArgumentDefinitions.READ_FILTER_LONG_NAME,
+            shortName = StandardArgumentDefinitions.READ_FILTER_SHORT_NAME,
+            doc = "Filters to apply to reads before analysis",
+            common = true,
+            optional = true)
+    public List<CommandLineReadFilter> enabledCommandLineFilters = new ArrayList<>();
```

Comment:
Somehow I got the idea that we wanted to support disabling of individual default filters (I may have gotten that idea from forum postings since it has been discussed there some). But we could do what you suggest. Currently the code that resolves the defaults with the command line filters has a comment that says that enabled wins if there is a conflict - but I picked that arbitrarily, and if we keep it we should probably log/warn when that happens. I'm open to suggestions.